python/ray/air/_internal/tensorflow_utils.py

```diff
@@ -13,18 +13,23 @@
 def convert_ndarray_to_tf_tensor(
     ndarray: np.ndarray,
     dtype: Optional[tf.dtypes.DType] = None,
+    ragged: bool = False,
 ) -> tf.Tensor:
     """Convert a NumPy ndarray to a TensorFlow Tensor.
 
     Args:
         ndarray: A NumPy ndarray that we wish to convert to a TensorFlow Tensor.
         dtype: A TensorFlow dtype for the created tensor; if None, the dtype will be
             inferred from the NumPy ndarray data.
+        ragged: If ``True``, return a ragged tensor instead of a normal tensor.
 
     Returns: A TensorFlow Tensor.
     """
     ndarray = _unwrap_ndarray_object_type_if_needed(ndarray)
```

Comment:
In the `batch_size=1`, there's no way to determine if the `ndarray` is ragge. So, we need to determine the raggedness from the type spec.